src/Aspire.Hosting/Extensions/ExecutableResourceBuilderExtensions.cs

```diff
@@ -24,4 +26,27 @@ public static IResourceBuilder<ExecutableResource> AddExecutable(this IDistribut
         var executable = new ExecutableResource(name, command, workingDirectory, args);
         return builder.AddResource(executable);
     }
+
+    /// <summary>
+    /// Adds annotation to <see cref="ExecutableResource" /> to support containerization during deployment.
+    /// </summary>
+    /// <typeparam name="T">Type of executable resource</typeparam>
+    /// <param name="builder">Resource builder</param>
+    /// <returns></returns>
+    public static IResourceBuilder<T> WithDockerfile<T>(this IResourceBuilder<T> builder) where T : ExecutableResource
+    {
+        return builder.WithAnnotation(
+            new ManifestPublishingCallbackAnnotation(writer => WriteExecutableAsDockerfileResource(writer, builder.Resource))
+            );
+    }
+
+    private static void WriteExecutableAsDockerfileResource(Utf8JsonWriter writer, ExecutableResource executable)
+    {
+        writer.WriteString("type", "dockerfile.v0");
+        writer.WriteString("path", Path.Combine(executable.WorkingDirectory, "Dockerfile"));
```

Comment:
This needs to be resolved to the relative path using all of the crazy logic in the manifest publisher. Maybe expose a method.

Reply:
Also, this should blow up if you don't have a dockerfile?

Reply:
I think I'm about to make the crazy breaking change to the callback :)

Reply:
arg. Can we do it in a non-breaking way? Support both, obsolete the old one?